### DISCUSSION TITLE
Even after loading private key for certificate, swanctl --list-certs not showing "has private key" field

### DISCUSSION THREAD

**Discussion by digambaringale333** [strongswan/strongswan#665] [Q&A]
Hi, We are using davici API's to load cert and associated private key in VICI daemon. In charon log's I could see that we have installed private key and certificate but swanctl --list-certs is not showing has_private key field.

Charon logs: Where we can see that cert and private key are loaded.
Oct 13 14:59:56 08[CFG] loaded certificate 'C=IN, ST=MH, L=Goa, O=ORG, OU=Systems, CN=cluster009, E=support@xyz.com'
Oct 13 14:59:56 16[CFG] loaded RSA private key

# swanctl --list-certs

List of X.509 End Entity Certificates

  subject:  "C=IN, ST=MH, L=Goa, O=ORG, OU=Systems, CN=cluster009, E=support@xyz.com"
  issuer:   "C=IN, ST=MH, L=Goa, O=ORG, OU=Systems, CN=cluster009, E=support@xyz.com"
  validity:  not before Oct 13 14:50:29 2021, ok
             not after  Oct 09 14:50:29 2036, ok (expires in 5474 days)
  serial:    61:66:f2:35
  flags:     serverAuth clientAuth self-signed
  authkeyId: f0:3e:63:f6:52:3f:05:46:81:63:b0:05:f9:e3:33:92:e3:0e:3e:37
  subjkeyId: f0:3e:63:f6:52:3f:05:46:81:63:b0:05:f9:e3:33:92:e3:0e:3e:37
  pubkey:    RSA 2048 bits
  keyid:     92:92:46:79:58:07:a5:0d:ee:4d:d6:79:38:96:4e:52:c1:e7:44:f9
  subjkey:   f0:3e:63:f6:52:3f:05:46:81:63:b0:05:f9:e3:33:92:e3:0e:3e:37

**Comment by tobiasbrunner** [comment]
The certificate and private key apparently don't match (based on a fingerprint over subjectPublicKey, i.e. _subjkey_).

**Reply by digambaringale333** [reply]
Interestingly, If restart strongswan after this(above mentioned issue) then it detects the private key. Do you think the way we load private key in VICI is important?
currently I am loading end point certificate first(using load_cert API) and the private key(using load_key API)

**Reply by tobiasbrunner** [reply]
> If restart strongswan after this(above mentioned issue) then it detects the private key

What does that mean? You load the cert and key, then restart without loading anything and it has both magically loaded and associated with each other?

> Do you think the way we load private key in VICI is important?

Perhaps (no idea what you are doing exactly). But e.g. the order does not matter.
